test-alerts.cpp

```diff
@@ -2,7 +2,76 @@
 
 #include "test/catch.hpp"
 #include "typewise-alert.h"
+#include "string.h"
 
-TEST_CASE("infers the breach according to limits") {
+
+BatteryCharacter batteryCharTest1,batteryCharTest2;
+extern TempFlags Tempflag;
+
+
+
+TEST_CASE("LOW limit value test") {
   REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
 }
+TEST_CASE("infers the breach according to limits") {
+  REQUIRE(inferBreach(42, 20, 30) == TOO_HIGH);
+}
+TEST_CASE("Normal value test") {
+  REQUIRE(inferBreach(25, 20, 30) == NORMAL);
+}
+TEST_CASE("Same value test") {
+  REQUIRE(inferBreach(20, 20, 20) == NORMAL);
+}
+TEST_CASE("F2_T1") {
+  REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 37) == TOO_HIGH);
+}
+TEST_CASE("F2_T2") {
+  REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, 34) == NORMAL);
+}
+TEST_CASE("F2_T3") {
+  REQUIRE(classifyTemperatureBreach(PASSIVE_COOLING, -34) == TOO_LOW);
+}
+
+TEST_CASE("F2_T7") {
+  REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 41) == TOO_HIGH);
+}
+TEST_CASE("F2_T8") {
+  REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, 35) == NORMAL);
+}
+TEST_CASE("F2_T9") {
+  REQUIRE(classifyTemperatureBreach(MED_ACTIVE_COOLING, -40) == TOO_LOW);
+}
+
+TEST_CASE("F2_T4") {
+  REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, 46) == TOO_HIGH);
+}
+TEST_CASE("F2_T5") {
+  REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, 34) == NORMAL);
+}
+TEST_CASE("F2_T6") {
+  REQUIRE(classifyTemperatureBreach(HI_ACTIVE_COOLING, -100) == TOO_LOW);
+}
+
+TEST_CASE("send email alert high") {
+batteryCharTest1.coolingType = PASSIVE_COOLING;
+checkAndAlert(TO_EMAIL,batteryCharTest1,50);
+REQUIRE(Tempflag.FlagHighTemp == TEMPHIGH);
```

Comment:
TempFlag is a global variable being used many times. If the values are not cleared before the next test case, then the probability of error is very high, for ex, if someone does a code refactor or rearranges the test cases in future the previous set values can be a problem for further tests!
This needs to be taken care (or atleast you should be aware of)...

Reply:
OK Got it i will update it.